spring-cloud-stream-binder-kinesis-core/src/main/java/org/springframework/cloud/stream/binder/kinesis/provisioning/KinesisStreamProvisioner.java

```diff
@@ -40,34 +40,40 @@
 		implements ProvisioningProvider<ExtendedConsumerProperties<KinesisConsumerProperties>,
 		ExtendedProducerProperties<KinesisProducerProperties>> {
 
-	private final AmazonKinesis amazonKinesis;
+	private final Log logger = LogFactory.getLog(getClass());
+
+	private final KinesisStreamHandler kinesisStreamHandler;
 
 	private final KinesisBinderConfigurationProperties configurationProperties;
 
-	public KinesisStreamProvisioner(AmazonKinesis amazonKinesis,
-			KinesisBinderConfigurationProperties kinesisBinderConfigurationProperties) {
-		Assert.notNull(amazonKinesis, "'amazonKinesis' must not be null");
+	public KinesisStreamProvisioner(KinesisStreamHandler kinesisStreamHandler,
+									KinesisBinderConfigurationProperties kinesisBinderConfigurationProperties) {
+		Assert.notNull(kinesisStreamHandler, "'kinesisStreamHandler' must not be null");
 		Assert.notNull(kinesisBinderConfigurationProperties, "'kinesisBinderConfigurationProperties' must not be null");
-		this.amazonKinesis = amazonKinesis;
+		this.kinesisStreamHandler = kinesisStreamHandler;
 		this.configurationProperties = kinesisBinderConfigurationProperties;
 	}
 
 	@Override
 	public ProducerDestination provisionProducerDestination(String name,
 			ExtendedProducerProperties<KinesisProducerProperties> properties) throws ProvisioningException {
+		logger.info("Using Kinesis stream for outbound: " + name);
 
-		KinesisProducerDestination producer = new KinesisProducerDestination(name);
+		KinesisStream stream = kinesisStreamHandler.createOrUpdate(name, 1,
```

Comment:
As @artembilan suggested, instead of hard-coding the value `1` for the shard counts, get it from the provided producer and consumer properties.